mongo/integration/unified/client_entity.go

```diff
@@ -31,14 +31,36 @@ type clientEntity struct {
 	succeeded       []*event.CommandSucceededEvent
 	failed          []*event.CommandFailedEvent
 	ignoredCommands map[string]struct{}
+
+	// These should not be changed after the clientEntity is initialized
+	storePoolCreated               []string
+	storePoolReady                 []string
+	storePoolCleared               []string
+	storePoolClosed                []string
+	storeConnectionCreated         []string
+	storeConnectionReady           []string
+	storeConnectionClosed          []string
+	storeConnectionCheckOutStarted []string
+	storeConnectionCheckOutFailed  []string
+	storeConnectionCheckedOut      []string
+	storeConnectionCheckedIn       []string
+	storeCommandStarted            []string
+	storeCommandSucceeded          []string
+	storeCommandFailed             []string
```

Comment:
A lot of fields; I wonder if we could just map event type strings (e.g. `CommandStartedEvent`) to arrays of ID strings.